### PR TITLE
added hasOwnProperty checks to all for/in loops

### PR DESCRIPTION
**Pull request by briangoss**
This protects the loops from crashing if the app or another module adds to the Object prototype.

### CODE REVIEW

File: lib/http.js [hunk 1]
```diff
@@ -73,12 +75,16 @@ HttpClient.prototype.buildRequest = function(rurl, data, exheaders, exoptions) {
 
   exoptions = exoptions || {};
   for (attr in exoptions) {
-    if (mergeOptions.indexOf(attr) !== -1) {
-      for (header in exoptions[attr]) {
-        options[attr][header] = exoptions[attr][header];
+    if (exoptions.hasOwnProperty(attr)) {
+      if (mergeOptions.indexOf(attr) !== -1) {
```
Comment (herom) [inline lib/http.js]: Could you please combine both checks as we don't have an `else` for the second condition?
(`if (exoptions.hasOwnProperty(attr) && mergeOptions.indexOf(attr) !== -1)`)

File: lib/nscontext.js [hunk 2]
```diff
@@ -96,8 +96,10 @@ NamespaceScope.prototype.getPrefix = function(nsUri, localOnly) {
       return 'xmlns';
     default:
       for (var p in this.namespaces) {
-        if (this.namespaces[p].uri === nsUri) {
-          return p;
+        if (this.namespaces.hasOwnProperty(p)) {
+          if (this.namespaces[p].uri === nsUri) {
+            return p;
```
Comment (herom) [inline lib/nscontext.js]: could you please combine these two conditionals too?

File: lib/wsdl.js [hunk 3]
```diff
@@ -459,9 +463,11 @@ MessageElement.prototype.postProcess = function(definitions) {
   var type;
 
   for (i in children) {
-    if ((child = children[i]).name === 'part') {
-      part = child;
-      break;
+    if (children.hasOwnProperty(i)) {
+      if ((child = children[i]).name === 'part') {
+        part = child;
```
Comment (herom) [inline lib/wsdl.js]: please combine these two conditionals 😸

File: lib/wsdl.js [hunk 4]
```diff
@@ -523,7 +529,9 @@ MessageElement.prototype.postProcess = function(definitions) {
 
       if (typeNs) {
         if (type.name in Primitives) {
+          if (Primitives.hasOwnProperty(type.name)) {
             // this.element = this.element.$type;
```
Comment (herom) [inline lib/wsdl.js]: I think we just found some "dead" code here so I don't think we need a guard here but rather look what caused this code to be commented out and left behind 🌵

File: lib/wsdl.js [hunk 5]
```diff
@@ -787,7 +795,9 @@ ExtensionElement.prototype.description = function(definitions, xmlns) {
       schema = definitions.schemas[ns];
 
     if (typeName in Primitives) {
-      return this.$base;
+      if (Primitives.hasOwnProperty(typeName)) {
```
Comment (herom) [inline lib/wsdl.js]: please combine (sorry for keeping on trolling you...)

File: lib/wsdl.js [hunk 6]
```diff
@@ -1035,34 +1055,42 @@ var WSDL = function(definition, uri, options) {
...
+          if (typeof binding.style === 'undefined') {
+            binding.style = 'document';
+          }
+          if (binding.style !== 'document')
+            continue;
+          var methods = binding.methods;
+          var topEls = binding.topElements = {};
+          for (var methodName in methods) {
+            if (methods.hasOwnProperty(methodName)) {
+              if (methods[methodName].input) {
```
Comment (herom) [inline lib/wsdl.js]: please combine these two too 👍

**Comment by herom** [comment]
Do you plan on integrating the requested changes @briangoss ? If you need some further info/clarification please don't hesitate to ask 👍

**Comment by briangoss** [comment]
Yes but it may take me a while to find the time. Sorry for the delay.

**Comment by herom** [comment]
no problem - I just wanted to make sure that this PR isn't orphaned 👍 😃

**Comment by Kankan023** [comment]
Hi, 
Is this issue solved? I am unable to see the hasOwnProperty check in my code that I have recently downloaded. If not I can help resolving this PR. 
Let me know
Thanks

**Pull request closed without merging**